src/test/java/github/weichware10/util/db/TrialsTest.java

```diff
@@ -0,0 +1,309 @@
+
+        TrialData trialData16 = new TrialData(ToolType.CODECHARTS, trialsCc.get(2),
+                dbClient.configurations.setConfiguration(codeConfig));
+        trialData16.setAnswer("Ja!");
+        assertFalse(dbClient.trials.setTrial(trialData16));
+    }
+
+    @Test
+    public void getTrialShouldWork() {
+        List<String> trialsZm = dbClient.trials.addTrials(configIdZm, 1);
+
+        TrialData trialData1 = new TrialData(ToolType.ZOOMMAPS, trialsZm.get(0), configIdZm);
+        trialData1.setAnswer("Ja");
+        trialData1.addDataPoint(new int[] { 1, 1 }, 1.0f);
+        trialData1.addDataPoint(new int[] { 2, 2 }, 1.0f);
+        trialData1.addDataPoint(new int[] { 3, 3 }, 1.0f);
+        dbClient.trials.setTrial(trialData1);
+        TrialData trialData11 = dbClient.trials.getTrial(trialsZm.get(0));
+        assertTrue(trialData11.equals(trialData1));
+
+        List<String> trialsCc = dbClient.trials.addTrials(configIdCc, 1);
+
+        TrialData trialData2 = new TrialData(ToolType.CODECHARTS, trialsCc.get(0), configIdCc);
+        trialData2.setAnswer("Ja");
+        trialData2.addDataPoint(new int[] { 1, 1 }, new int[] { 2, 2 });
+        trialData2.addDataPoint(new int[] { 2, 2 }, new int[] { 3, 3 });
+        trialData2.addDataPoint(new int[] { 3, 3 }, new int[] { 4, 4 });
+        dbClient.trials.setTrial(trialData2);
+        TrialData trialData22 = dbClient.trials.getTrial(trialsCc.get(0));
+        assertTrue(trialData22.equals(trialData2));
+    }
+
+    @Test
+    public void getTrialListShouldWork() {
+        // ZOOMMAPS
+        List<String> trialsZm = dbClient.trials.addTrials(configIdZm, 3);
+        TrialData trialData1 = new TrialData(ToolType.ZOOMMAPS, trialsZm.get(0), configIdZm,
+                new DateTime(2021, 12, 12, 12, 0, 0), "Ja", new ArrayList<DataPoint>());
+        trialData1.addDataPoint(new int[] { 1, 1 }, 1.0f);
+        dbClient.trials.setTrial(trialData1);
+
+        TrialData trialData2 = new TrialData(ToolType.ZOOMMAPS, trialsZm.get(1), configIdZm,
+                new DateTime(2021, 12, 12, 13, 0, 0), "Ja", new ArrayList<DataPoint>());
+        trialData2.addDataPoint(new int[] { 2, 2 }, 2.0f);
+        dbClient.trials.setTrial(trialData2);
+
+        TrialData trialData3 = new TrialData(ToolType.ZOOMMAPS, trialsZm.get(2), configIdZm,
+                new DateTime(2021, 12, 13, 14, 0, 0), "Ja", new ArrayList<DataPoint>());
+        trialData3.addDataPoint(new int[] { 3, 3 }, 3.0f);
+        dbClient.trials.setTrial(trialData3);
+
+        String configIdZm2 = dbClient.configurations.setConfiguration(zoomConfig);
+        List<String> trialsZm2 = dbClient.trials.addTrials(configIdZm2, 1);
+
+        TrialData trialData7 = new TrialData(ToolType.ZOOMMAPS, trialsZm2.get(0), configIdZm2,
+                new DateTime(2021, 12, 10, 14, 0, 0), "Ja", new ArrayList<DataPoint>());
+                trialData7.addDataPoint(new int[] { 4, 4 }, 4.0f);
```

Comment:
⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.indentation.IndentationCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'method def' child has incorrect indentation level 16, expected level should be 8.